SQL_assesment.txt

```diff
@@ -0,0 +1,1829 @@
+| 2006 | Clemens | London |    100 | 1001 |
+| 2007 | Pereira | Rome   |    100 | 1004 |
++------+---------+--------+--------+------+
+
+
+3) Find the largest order taken by each salesperson on each date.
+
+mysql> SELECT SNUM, ODATE, MAX(AMT) AS "Maximum Amount"
+    -> FROM Orders
+    -> GROUP BY
+    -> ODATE,ONUM;
+
++------+------------+----------------+
+| SNUM | ODATE      | Maximum Amount |
++------+------------+----------------+
+| 1007 | 1990-10-03 |          18.69 |
+| 1004 | 1990-10-03 |         1900.1 |
+| 1001 | 1990-10-03 |         767.19 |
+| 1002 | 1990-10-03 |        5160.45 |
+| 1007 | 1990-10-03 |        1098.16 |
+| 1002 | 1990-10-04 |          75.75 |
+| 1001 | 1990-10-05 |           4723 |
+| 1003 | 1990-10-04 |        1713.23 |
+| 1002 | 1990-10-06 |        1309.95 |
+| 1001 | 1990-10-06 |        9891.88 |
++------+------------+----------------+
+
+
+4) Arrange the Order table by descending customer number.
+
+mysql> SELECT ONUM, AMT, ODATE, CNUM, SNUM
+    -> FROM Orders
+    -> ORDER BY CNUM DESC;
+
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-10-03 | 2008 | 1007 |
+| 3006 | 1098.16 | 1990-10-03 | 2008 | 1007 |
+| 3002 |  1900.1 | 1990-10-03 | 2007 | 1004 |
+| 3008 |    4723 | 1990-10-05 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-10-06 | 2006 | 1001 |
+| 3007 |   75.75 | 1990-10-04 | 2004 | 1002 |
+| 3010 | 1309.95 | 1990-10-06 | 2004 | 1002 |
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-10-04 | 2002 | 1003 |
+| 3003 |  767.19 | 1990-10-03 | 2001 | 1001 |
++------+---------+------------+------+------+
+
+
+5) Find which salespeople currently have orders in the order table. 
+
+mysql> SELECT SNUM,SNAME FROM Salespeople
+    -> WHERE 
+    -> SNUM = ANY(SELECT DISTINCT(SNUM) FROM Orders);
+
++------+---------+
+| SNUM | SNAME   |
++------+---------+
+| 1001 | Peel    |
+| 1002 | Serres  |
+| 1003 | AxelRod |
+| 1004 | Motika  |
+| 1007 | Rifkin  |
++------+---------+
+
+
+6) List names of all customers matched with the salespeople serving them. 
+mysql> SELECT CNAME, SNAME FROM Customers c, salespeople s
+    -> WHERE
+    -> s.SNUM = c.snum ;
++----------+---------+
+| CNAME    | SNAME   |
++----------+---------+
+| Hoffman  | Peel    |
+| Giovanni | AxelRod |
+| Liu      | Serres  |
+| Grass    | Serres  |
+| Clemens  | Peel    |
+| Cisneros | Rifkin  |
+| Pereira  | Motika  |
++----------+---------+
+
+
+7) Find the names and numbers of all salespeople who have more than one customer. 
+
+mysql>  SELECT SNAME, SNUM FROM Salespeople
+    ->  WHERE 
+    ->  SNUM =  ANY(SELECT SNUM FROM Customers
+    ->  GROUP BY SNUM 
+    ->  HAVING 
+    ->  COUNT(SNUM)>1);
+
++--------+------+
+| SNAME  | SNUM |
++--------+------+
+| Peel   | 1001 |
+| Serres | 1002 |
++--------+------+
+
+
+8) Count the orders of each of the salespeople and output the results in descending order. 
```

Comment:
What if salespeople are not associated with any order? There should be a zero for him/her.